### PR TITLE
[macOS][NativeWindowing] Fix fullscreen window menu item (and shortcuts)

### PR DESCRIPTION
**Pull request by enen92**
## Description
This essentially provides two fixes:
- Fixes the toggle fullscreen app bar menu item (was actually unimplemented):
image

- The FN/Globe+f shortcut (displayed in the menu as the `keyEquivalent` for fullscreenToggle as the result of f + `NSEventModifierFlagCommand | NSEventModifierFlagControl` was not being trapped as a shortcut in `ProcessOSXShortcuts`. Since it works on all other apps and may lead to user confusion, it's implemented on a separate commit.

### CODE REVIEW

**Comment by enen92** [comment]
Merging by the end of the weekend unless someone opposes.

File: xbmc/windowing/osx/WinEventsOSXImpl.mm [hunk 1]
```diff
@@ -145,6 +148,7 @@ - (bool)ProcessOSXShortcuts:(XBMC_Event&)event
         return true;
 
       case XBMCK_CTRLF: // CMD-CTRL-f to toggle fullscreen
+      case XBMCK_f: // FN/Globe-f to toggle fullscreen
```
Comment (kambala-decapitator) [inline xbmc/windowing/osx/WinEventsOSXImpl.mm:151]: if I understand correctly, now cmd+F will also enter fullscreen. Not sure it's right.

still unclear why cmd+ctrl+F doesn't work (if I understood your words right).

btw in Safari menu on macOS 12.6 I also see Fn+F shortcut for Fullscreen, but cmd+ctrl+F still works as well

File: xbmc/windowing/osx/WinEventsOSXImpl.mm [hunk 2]
```diff
@@ -302,6 +306,7 @@ - (NSEvent*)InputEventHandler:(NSEvent*)nsevent
       [self MessagePush:&newEvent];
       passEvent = false;
       break;
+    case kCGEventFlagsChanged:
```
Comment (kambala-decapitator) [inline xbmc/windowing/osx/WinEventsOSXImpl.mm:309]: what is this for?

Comment (enen92) [reply xbmc/windowing/osx/WinEventsOSXImpl.mm:309]: ups, leftover...
thanks

File: xbmc/windowing/osx/WinEventsOSXImpl.mm (hunk 1 above)
Comment (enen92) [reply xbmc/windowing/osx/WinEventsOSXImpl.mm]: Nice point. This revision was also toggling fullscreen for cmd-f (did not notice that).
Updated in the last revision

File: xbmc/windowing/osx/WinEventsOSXImpl.mm [hunk 3]
```diff
@@ -147,7 +154,13 @@ - (bool)ProcessOSXShortcuts:(XBMC_Event&)event
...
         CServiceBroker::GetAppMessenger()->PostMsg(TMSG_TOGGLEFULLSCREEN);
         return true;
-
+      
+      case XBMCK_f: // FN/Globe-f to toggle fullscreen
+        if (isFn) // avoid cmd-f to toggle fullscreen
+        {
+          CServiceBroker::GetAppMessenger()->PostMsg(TMSG_TOGGLEFULLSCREEN);
+        }
+        return true;
```
Comment (kambala-decapitator) [inline xbmc/windowing/osx/WinEventsOSXImpl.mm:163]:
```suggestion
          CServiceBroker::GetAppMessenger()->PostMsg(TMSG_TOGGLEFULLSCREEN);
          return true;
        }
        break;
```

**Review by kambala-decapitator — APPROVED** [review]

**Pull request merged**